vendor/github.com/coreos/etcd/clientv3/concurrency/mutex.go

```diff
@@ -68,11 +68,10 @@ func (m *Mutex) Lock(ctx context.Context) error {
 
 	// wait for deletion revisions prior to myKey
 	hdr, werr := waitDeletes(ctx, client, m.pfx, m.myRev-1)
-	// release lock key if cancelled
-	select {
-	case <-ctx.Done():
+	// release lock key if wait failed
+	if werr != nil {
 		m.Unlock(client.Ctx())
-	default:
+	} else {
```

Comment:
I'd prefer to avoid the `else` clause and return explicitly in the error handling block. That makes the flow more readable IMO.

```suggestion
                return werr
	}
```

I'm not sure on the implications of removing the `select`. Was it only used as a sync method?